File: #083/Remove Duplicates from Sorted List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head==null)
            return head;
        HashSet<Integer> set = new HashSet<>();
        ListNode p = head;
        set.add(head.val);
        while (p != null && p.next != null) {
            if (set.contains(p.next.val)) {
                p.next = p.next.next;
            } else {
                set.add(p.next.val);
            p = p.next;
            }
        }
        return head;
    }
}

File: #083/Remove Duplicates from Sorted List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
	ListNode* deleteDuplicates(ListNode* head) {
		if (!head) return head;
		int last = head->val;
		ListNode *p = head;
		while (p != NULL&&p->next != NULL) {
			if (p->next->val == last)
				p->next = p->next->next;
			else {
				last = p->next->val;
				p = p->next;
			}
			
		}
		return head;
	}
};